client/jetpack-onboarding/index.js

```diff
@@ -0,0 +1,17 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import page from 'page';
+import { isEnabled } from 'config';
+
+/**
+ * Internal dependencies
+ */
+import { onboarding } from './controller';
+
+export default function() {
+	if ( isEnabled( 'jetpack/onboarding' ) ) {
+		page( '/jetpack/onboarding/:stepName?', onboarding );
```

Comment:
We could (and should 😄) add a whitelist for `stepName` here, cf #20422. Fine to do it in separate PR tho.

Reply:
I'd be happy to defer that to you 😆 TBH I'm not sure what's the rationale behind that.

Reply:
Okay, will do in a subsequent PR.

As for the rationale, we'll want to avoid matching routes for which we don't know what to do. Using a whitelist here, we can fall through to a 404 (or, if it's only on the client side, a simple 'Not Found' component) in all other cases.